src/Interfaces.ts

```diff
@@ -213,3 +213,15 @@ export type GraphQLParseOptions = {
   allowLegacySDLImplementsInterfaces?: boolean;
   experimentalFragmentVariables?: boolean;
 };
+
+type ResolvedScalar = string | number | boolean | null;
+type ResolvedValue = ResolvedScalar | ResolvedScalar[] | { [key: string]: ResolvedValue };
+type ResolverFunction = (...args: any[]) => ResolvedValue;
+
+export type ResolverMap = {
+  [key: string]: () =>
+    | { [key: string]: ResolvedValue | ResolverFunction }
+    | null
+    | ResolvedValue
+    | ResolverMap;
```

Comment:
the two typing changes were to add `| ResolvedValue | ResolverMap` here.

A ResolverMap value can be a ResolvedValue directly - no need for a ResolverFunction. This seems to work fine in our testing.

The other change is to allow a ResolverMap value to be another ResolverMap. The ResolverMap can be multilevel. For instance, one test we wrote passes:

```
    const client = mockClient({
      WorkItem: () => ({
        queue: () => ({
          id: () => 'test-queue-id-1',
        }),
      }),
    });
```
(here, mockClient is our local mock Apollo client and it calls mergeResolvers() with the first argument being our base set of mock resolvers for all tests, and the second argument being the test-specific resolver override map)